screens/UserScreens/UserDashboard.jsx

```diff
@@ -1,15 +1,54 @@
 import { Modal, View, Text, ScrollView } from "react-native";
-import CreateAndJoinButtons from "../../components/UserDashboard/CreateAndJoin";
-import ProductionButton from "../../components/UserDashboard/ProductionButton";
+import { useState, useEffect } from "react";
+import { onValue, get, ref, child, set } from "firebase/database";
 import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
 import { faClapperboard, faMasksTheater } from "@fortawesome/free-solid-svg-icons";
-import { mockProductions } from "../../MockData/MockProductions";
-import { useState } from "react";
+import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import CreateAndJoinButtons from "../../components/UserDashboard/CreateAndJoin";
+import ProductionButton from "../../components/UserDashboard/ProductionButton";
 import JoinProductionModal from "../../components/ProductionModals/JoinProductionModal";
 import CreateProductionModal from "../../components/ProductionModals/CreateProductionModal";
 
+//TODO:
+// - order participants by last action/ time joined
+
 export default function UserDashboardScreen({ navigation }) {
-    let [modal, setModal] = useState(null);
+    const [modal, setModal] = useState(null);
+    const [createName, setCreateName] = useState("");
+    const [joinCode, setJoinCode] = useState("");
+    const [productions, setProductions] = useState([]);
+    const auth = firebase_auth;
+    const db = firebase_db;
+
+    useEffect(() => {
+        onValue(ref(db, `users/${auth.currentUser.uid}/productions`), async (userSnapshot) => {
+            if (!userSnapshot.exists()) return;
+            const userData = userSnapshot.val();
+
+            const newProds = await Promise.all(
+                Object.keys(userData).map(async (productionCode) => {
+                    return get(child(ref(db), `productions/${productionCode}`))
+                        .then((playSnapshot) => {
+                            if (!playSnapshot.exists()) return;
+                            const playData = playSnapshot.val();
```

Comment:
Would maybe be better if there was an Activity Indicator instead of saying No Plays Found whilst this async thing is going on, then after the promise all set it to the text if there's no data

Reply:
having a loading state set at the start of the useeffect and then set to false at the end doesnt really work because the state change triggers the useeffect again so it perpetually loads. I've found another way to do it but up to you if you think its janky lol

Reply:
now done with loading state 👍